hws/src/dr_device_resolver.py

```diff
@@ -0,0 +1,356 @@
+    except Exception:
+        pass
+
+    return None
+
+
+def resolve_device(device_str):
+    """
+    Main function to resolve device identifier to all formats.
+
+    Args:
+        device_str: Device identifier in any supported format
+
+    Returns:
+        DeviceIdentifier object with resolved values
+
+    Raises:
+        ValueError: If device string cannot be resolved
+    """
+    device_type = _detect_device_type(device_str)
+
+    if device_type is None:
+        raise ValueError(f"Unable to detect device type for: {device_str}")
+
+    pci_bdf = None
+    rdma_dev = None
+    mst_dev = None
+
+    if device_type == 'mst':
+        # MST device provided
+        mst_dev = device_str
+        pci_bdf = _resolve_pci_from_mst(mst_dev)
+        if pci_bdf:
+            rdma_dev = _get_rdma_from_pci(pci_bdf)
+        else:
+            # Try to get RDMA directly from mst status
+            rdma_dev = _get_rdma_from_mst(mst_dev)
+            if rdma_dev:
+                # Try to get PCI from RDMA
+                pci_bdf = _resolve_pci_from_sysfs(rdma_dev, 'ib')
+
+    elif device_type == 'pci':
+        # PCI BDF provided
+        pci_bdf = _normalize_pci_bdf(device_str)
+        rdma_dev = _get_rdma_from_pci(pci_bdf)
+        mst_dev = _get_mst_from_pci(pci_bdf)
+
+    elif device_type == 'ib':
+        # InfiniBand device provided
+        rdma_dev = device_str
+        pci_bdf = _resolve_pci_from_sysfs(device_str, 'ib')
+        if pci_bdf:
+            mst_dev = _get_mst_from_pci(pci_bdf)
+
+    elif device_type == 'netdev':
+        # Network device provided
+        pci_bdf = _resolve_pci_from_sysfs(device_str, 'netdev')
+        if pci_bdf:
+            rdma_dev = _get_rdma_from_pci(pci_bdf)
+            mst_dev = _get_mst_from_pci(pci_bdf)
+
+    # Validate that we at least have PCI BDF
+    if pci_bdf is None:
+        raise ValueError(f"Unable to resolve PCI BDF for device: {device_str}")
+
+    return DeviceIdentifier(pci_bdf=pci_bdf, rdma_dev=rdma_dev, mst_dev=mst_dev)
+
+
+def get_device_for_resourcedump(device_str):
+    """
+    Convenience function to get the best device identifier for resourcedump.
+
+    Resourcedump can accept:
+    - PCI BDF (most universal)
+    - MST device path
+
+    This function returns PCI BDF as it's the most universal format.
+
+    Args:
+        device_str: Device identifier in any supported format
+
+    Returns:
+        String suitable for resourcedump -d parameter
+    """
+    dev_id = resolve_device(device_str)
+
+    # Prefer PCI BDF as it's most universal
+    if dev_id.pci_bdf:
+        return dev_id.pci_bdf
+
+    # Fallback to MST device if available
+    if dev_id.mst_dev:
+        return dev_id.mst_dev
+
+    raise ValueError(f"Unable to determine resourcedump device for: {device_str}")
+
+
+if __name__ == "__main__":
+    # Simple test/demo
+    import sys
+    if len(sys.argv) < 2:
+        print("Usage: python dr_device_resolver.py <device>")
+    print("\nExamples:")
```

Comment:
get_device_for_resourcedump function and the whole __main__ part: do we want to leave this in the code?
I understand that this was important during the development, but I'm not sure it should stay.
In any case, see below an indentation bug.